Assets/Global/Scripts/Player/States/AttackingState.cs

```diff
@@ -1,158 +1,70 @@
-using System.Data;
-using System.Runtime.InteropServices.WindowsRuntime;
-using JetBrains.Annotations;
-using NUnit.Framework;
-using UnityEditor;
+using Unity.VisualScripting;
 using UnityEngine;
 using UnityEngine.InputSystem;
-
+using System.Linq;
 public class AttackingState : StateBase
 {
-    public bool rotateLeft;
-
-    public float rotate;
-
-    private bool isReturning;
-
-    private bool turnLeft;
-
     public AttackingState(Player player)
         : base(player) { }
 
-    public override void FixedUpdate()
-    {
-        Attack(Player.attackCounter);
-    }
-
-    void Attack(int attackCounter)
-    {
-        switch (attackCounter)
-        {
-            case 1:
-                Slash();
-                break;
-            case 2:
-                Slash();
-                break;
-            case 3:
-                GroundPound();
-                break;
-        }
-    }
-
-    void GroundPound()
+    public void IncreaseIndex()
     {
-        if (Player.isGrounded)
-        {
-            Player.isSlamming = true;
-            Player.rb.AddForce(Vector3.up * Player.slamForce, ForceMode.Impulse);
-        }
-        else if (Player.rb.linearVelocity.y < 0 && Player.isSlamming)
-        {
-            Player.rb.AddForce(Vector3.down * Player.slamForce, ForceMode.Impulse);
-        }
+        Player.attackIndex =
+            Player.attackIndex >= Player.Tail.GetComponent<Tail>().currentTail.currentCombo.Count - 1
+                ? 0
+                : ++Player.attackIndex;
     }
 
-    void Slash()
+    public override void Enter()
     {
-        float speed = turnLeft ?
-            Player.playerStatistic.AttackSpeedMultiplier.GetValue() :
-            -Player.playerStatistic.AttackSpeedMultiplier.GetValue();
-            
-        if (!isReturning)
+        var tail = Player.Tail.GetComponent<Tail>().currentTail;
```

Comment:
maybe personal preference. but i'd prefer is we explicitly cast our type rather than using var.
I assume we know this will always be of type BaseTail